runtime/lua/vim/lsp/tagfunc.lua

```diff
@@ -47,20 +46,21 @@ local function query_workspace_symbols(pattern)
     return {}
   end
   local results = {}
-  for _, symbols in pairs(results_by_client) do
+  for client_id, symbols in pairs(results_by_client) do
+    local client = lsp.get_client_by_id(client_id)
     for _, symbol in pairs(symbols.result or {}) do
       local loc = symbol.location
-      local item = mk_tag_item(symbol.name, loc.range, loc.uri)
+      local item = mk_tag_item(symbol.name, loc.range, loc.uri, client.offset_encoding)
       item.kind = lsp.protocol.SymbolKind[symbol.kind] or 'Unknown'
       table.insert(results, item)
     end
   end
   return results
 end
 
-function M.tagfunc(pattern, flags)
+local function tagfunc(pattern, flags)
   local matches
-  if flags == 'c' then
+  if string.match(flags, 'c') then
     matches = query_definition(pattern)
   elseif flags == '' or flags == 'i' then
```

Comment:
need string.match here too

```suggestion
  elseif flags == '' or string.match(flags, 'i')
```

@mfussenegger what is the case that flags is an empty string? the help doc on `tag-function` doesn't seem to mention this as a case. Should empty `flags` just return `vim.NIL`?

Reply:
Done on my end

Reply:
> what is the case that flags is an empty string? the help doc on tag-function doesn't seem to mention this as a case. Should empty flags just return vim.NIL?

If I invoke `:tags foo` the flags are empty for me.

Reply:
Ok that makes sense: `'c'` is for a normal-mode command (e.g. `<C-]>` or `g]`), `'i'` is for insert mode completion, and an empty flag is for an ex command.